samples/net/dhcpv4_client/README.rst

```diff
@@ -1,7 +1,8 @@
-.. _dhcpv4-client-sample:
+.. zephyr:code-sample:: dhcpv4-client
+   :name: DHCPv4 client application
```

Comment:
I checked how the documenation page looks like, and to my eyes the "application" here and in couple of other places in sample apps page, look unnecessary so I suggest we remove that. So this would just say "DHCPv4 client"

Reply:
absolutely, this was an oversight (or rather an half-baked copy-paste), as I am indeed trying really hard to ban "sample" or "application" from the samples' names